### PR TITLE
Cargo manifests cleanup

### PR DESCRIPTION
**Pull request by CBenoit**
~~This enables us to publish on crates.io~~
This is a first step towards enabling us to publish on crates.io.

I also ran [cargo-diet](https://github.com/the-lean-crate/cargo-diet), a helper for computing the optimal `include` directives for Cargo.toml manifests.

See #42

### CODE REVIEW

File: helix-term/Cargo.toml [hunk 1]
```diff
@@ -5,6 +5,7 @@ description = "A post-modern text editor."
 authors = ["Blaž Hrastnik <blaz@mxxn.io>"]
 edition = "2018"
 license = "MPL-2.0"
+include = ["src/**/*", "README.md"]
```
Comment (archseer) [inline helix-term/Cargo.toml:8]: I think this needs to include runtime/ somewhere so it can be installed with embed_runtime

Comment (CBenoit) [reply helix-term/Cargo.toml:8]: You're right!

**Comment by CBenoit** [comment]
@archseer this is not enough to publish on crates.io, but it is a least a first step.
For some reason I can't get cargo to package all the languages in `helix-syntax`. Of course we will have to do something to include the runtime in `helix-term` as well, but I'm not sure what is the right solution immediately (symlink?).

File: helix-syntax/Cargo.toml [hunk 2]
```diff
@@ -4,8 +4,11 @@ version = "0.2.0"
 authors = ["Blaž Hrastnik <blaz@mxxn.io>"]
 edition = "2018"
 license = "MPL-2.0"
-
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+description = "Tree-sitter grammars support"
+categories = ["editor"]
+repository = "https://github.com/helix-editor/helix"
+homepage = "https://helix-editor.com"
+include = ["src/**/*", "languages/**/*", "build.rs", "!**/docs/**/*", "!**/test/**/*", "!**/examples/**/*", "!**/build/**/*"]
```
Comment (pickfire) [inline helix-syntax/Cargo.toml:11]: I don't think we need to manually specify `include` right?

**Comment by pickfire** [comment]
@CBenoit How did you test out the publish?

File: helix-syntax/Cargo.toml (hunk 2 above)
Comment (CBenoit) [reply helix-syntax/Cargo.toml:11]: This is unrelated to docs.rs but it fact cargo _will_ package more than strictly necessary yes.
See https://github.com/the-lean-crate/cargo-diet

**Comment by CBenoit** [comment]
> @CBenoit How did you test out the publish?

I didn't test publish, I only test `cargo package` in `helix-syntax`.
I thought I could go farther, but as said…

**Review by archseer — APPROVED** [review]

**Pull request merged**